### DISCUSSION TITLE
Write word for ic2

### DISCUSSION THREAD

**Discussion by jedhutto** [Rubberazer/JETGPIO#13] [Ideas]
First off, thank you for writing this and making this public! 
I've very recently switched from raspberry pi (PIGPIO) to Jetson nano for my project which includes a PCA9685 for controlling servos. For that, I am to write by word through ic2 as well as by byte. So my question is, how difficult  would it be to write a method to write by word?

**Comment by Rubberazer** [comment]
Hi there.

It should be easy, I am kind of busy at work but will try to have a go next weekend. Will write a follow up comment when I have it ready.
On a side note, if you are starting with a PCA9685 you can have a look at this little [project](https://github.com/Rubberazer/Darkpaw_motion_opencv/blob/main/4_darkpaw.cpp) in case and find the i2cWriteByteData() functions, I am controlling 12 servos plugged to a PCA9685 with a similar library for Raspberry byte by byte and it was OK. I am using the same function names so it could be copy and paste. Maybe helpful if you are in a rush.

**Comment by Rubberazer** [comment]
I added 2 new functions, names are self descriptive, they work exactly the same as the byte versions but instead of writing/reading 1 byte, they do 2 bytes:

i2cWriteWordData()
i2cReadWordData()

Haven't done comprehensive testing just ran a quick one on reading, but haven't tried the write one yet.

**Reply by jedhutto** [reply]
Awesome, thanks! I really appreciate it!

**Comment by remilamoureuxlevesque** [comment]
In the same way, is it possible to add a functions to write/read X amount of bytes to one register by utilizing repeated start ?
